homeassistant/components/demo/light.py

```diff
@@ -30,33 +31,34 @@
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the demo light platform."""
-    async_add_entities(
-        [
-            DemoLight(
-                unique_id="light_1",
-                name="Bed Light",
-                state=False,
-                available=True,
-                effect_list=LIGHT_EFFECT_LIST,
-                effect=LIGHT_EFFECT_LIST[0],
-            ),
-            DemoLight(
-                unique_id="light_2",
-                name="Ceiling Lights",
-                state=True,
-                available=True,
-                ct=LIGHT_TEMPS[1],
-            ),
-            DemoLight(
-                unique_id="light_3",
-                name="Kitchen Lights",
-                state=True,
-                available=True,
-                hs_color=LIGHT_COLORS[1],
-                ct=LIGHT_TEMPS[0],
-            ),
-        ]
-    )
+    lights = [
+        DemoLight(
+            unique_id="light_1",
+            name="Bed Light",
+            state=False,
+            available=True,
+            effect_list=LIGHT_EFFECT_LIST,
+            effect=LIGHT_EFFECT_LIST[0],
+        ),
+        DemoLight(
+            unique_id="light_2",
+            name="Ceiling Lights",
+            state=True,
+            available=True,
+            ct=LIGHT_TEMPS[1],
+        ),
+        DemoLight(
+            unique_id="light_3",
+            name="Kitchen Lights",
+            state=True,
+            available=True,
+            hs_color=LIGHT_COLORS[1],
+            ct=LIGHT_TEMPS[0],
+        ),
+    ]
+    async_add_entities(lights)
+    data = hass.data.setdefault(DOMAIN, {})
+    data[LIGHT_DOMAIN] = lights
```

Comment:
We shouldn't make changes to integrations for tests. You can use `async_set` combined with capturing the services to make sure that you always get the right info.